homeassistant/components/icloud/config_flow.py

```diff
@@ -101,6 +105,17 @@ def __init__(self):
         if self.api.requires_2sa:
             return await self.async_step_trusted_device()
 
+        try:
+            devices = await self.hass.async_add_executor_job(
+                getattr, self.api, "devices"
+            )
+            if not devices:
+                raise PyiCloudNoDevicesException()
+        except (KeyError, PyiCloudNoDevicesException):
```

Comment:
Can the `KeyError` still happen?

Reply:
With the dependency bump, the `api. devices` error is catched now, and raise a `PyiCloudServiceNotActivatedException`.

Can still happen.